libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -378,12 +378,17 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 			if (_variable.isConstructorParameter())
 				errorString += " for constructor parameter";
 			else if (_variable.isCallableOrCatchParameter())
-				errorString +=
-					" for " +
-					string(_variable.isReturnParameter() ? "return " : "") +
-					"parameter in" +
-					string(_variable.isExternalCallableParameter() ? " external" : "") +
-					" function";
+			{
+							string visibility;
+							if (auto const* callable = dynamic_cast<CallableDeclaration const*>(_variable.scope()))
+								visibility = Declaration::visibilityToString(callable->visibility());
+							errorString +=
+							" for " +
+							string(_variable.isReturnParameter() ? "return " : "") +
+							"parameter in" +
+							string(visibility.empty() ? visibility : " " + visibility) +
```

Comment:
Since we're touching this code, we could also change the string conversion to a more concise one:
```suggestion
							_variable.isReturnParameter() ? "return "s : ""s +
							"parameter in" +
							visibility.empty() ? visibility : " "s + visibility +
```